test/asynchronous/test_cursor.py

```diff
@@ -1380,6 +1380,17 @@ async def test_getMore_does_not_send_readPreference(self):
         self.assertEqual("getMore", started[1].command_name)
         self.assertNotIn("$readPreference", started[1].command)
 
+    @async_client_context.require_sync
+    def test_to_list_when_empty(self):
```

Comment:
This test should cover calling to_list on tailable cursors. See https://pymongo.readthedocs.io/en/latest/examples/tailable.html#tailable-cursors for an example.

I still believe the current version of to_list will incorrectly loop forever on a tailable cursor.

Reply:
Ah, got it. The following test passes:

```
    async def test_to_list_tailable(self):
        self.db.drop_collection("test")

        await self.db.test.insert_many([{"x": i} for i in range(100)])

        c = self.db.test.find(cursor_type=pymongo.CursorType.TAILABLE_AWAIT)

        docs = await c.to_list()

        self.assertEqual(100, len(docs))  # type: ignore[arg-type]
```

Reply:
Sorry you have to create a tailable, await data cursor that yields an infinite number of documents, like tailing the oplog or running $chanegStream.

Reply:
Do we support running `$changeStream` separately from using our internal `change_stream` cursor? I see that cursor also has its own behavior for `try_next` that would need to be implemented in its own `_next_batch` method.

Reply:
This test using `oplog` also passes: 

```
    async def test_to_list_tailable(self):
        client = await async_rs_or_single_client()
        self.addAsyncCleanup(client.aclose)

        c = client.local.oplog.rs.find(
                   cursor_type=pymongo.CursorType.TAILABLE_AWAIT,
                   oplog_replay=True)

        docs = await c.to_list()

        self.assertGreater(len(docs), 0)  # type: ignore[arg-type]
```

Reply:
It's possible to run $changeStream directly via aggregate().